src/scrot.c

```diff
@@ -644,11 +644,10 @@ static void scrotExecApp(Imlib_Image image, struct tm *tm, char *filenameIM,
 {
     char *execStr = imPrintf(opt.exec, tm, filenameIM, filenameThumb, image);
     int ret = system(execStr);
-
     if (ret == -1)
-        warn("The child process could not be created");
-    else if (WIFEXITED(ret) && WEXITSTATUS(ret) == 127)
-        warnx("scrot could not execute the command: %s", execStr);
+        err(EXIT_FAILURE, "The child process could not be created");
+    else if (!WIFEXITED(ret) || WEXITSTATUS(ret) != 0)
+        exit(69 /* EX_UNAVAILABLE */ );
```

Comment:
It can't be merged as is because of https://github.com/resurrecting-open-source-projects/scrot/issues/271#issuecomment-1734754971
POSIX only defines what WEXITSTATUS() does if WIFEXITED() returns true.

Reply:
I'm a bit baffled here. If `WIFEXITED` was false then the exit status macro will not be evaluated due to the logical OR. What am I missing?

Reply:
That's because I'm a bad reader.
Still, it should be:
```c
else if (WIFEXITED(ret) && WEXITSTATUS(ret) || WTERMSIG(ret) && WIFSIGNALED(ret)
```

Reply:
>  Still, it should be:

Is there any functional difference here or just stylistic? AFAICS, the `!WIFEXITED(..)` should already catch `WIFSIGNALED(..)` case.

Reply:
Hmm, I see. I've made the check more explicit.

> FreeBSD has a wait4() function that returns if the child hits a breakpoint in a debugger

On a sidenote: Shouldn't this be considered a bug in freebsd if system returns without the child terminating?

Reply:
> > FreeBSD has a wait4() function that returns if the child hits a breakpoint in a debugger
> 
> On a sidenote: Shouldn't this be considered a bug in freebsd if system returns without the child terminating?

Yes.

Reply:
Hmm, actually, the `WTERMSIG(ret)` bit should be removed because it won't be optimized out and `0` is the null signal. The null signal can't happen.

Reply:
After further research and writing a test program, it appears that my interpretation of the way FreeBSD's `ptrace()` syscall works is wrong.

If a process doesn't use the `ptrace()` syscall, `system()` will never return because the child stopped. Only a process that is using the `ptrace()` API can have a `wait*()` function return `WIFSTOPPED()` even if the `WUNTRACED` flag wasn't specified. scrot doesn't use `ptrace()` and never will because that's for writing debuggers.

Reply:
So what's the conclusion? IMO the first version was short and read nicely: "If it wasn't a normal exit or the exit code was non-zero then exit with EX_UNAVAILABLE."